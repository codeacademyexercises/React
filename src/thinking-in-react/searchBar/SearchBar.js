import React, { Component } from 'react';
import '../App/App.css';

class SearchBar extends Component{
    render(){
        // console.log(this.props.search);
        return(
            <div className="SearchBarContainer">
                <div className="SearchBar">
                    <input type="search" placeholder="Search..." id="searchBar" defaultValue={this.props.search} onChange={this.props.SearchInput}/>
                </div>
                <div className="checkbox">
                    <input type="checkbox" id="inStock" defaultValue={this.props.stock} onChange={this.props.InStock}/>
                    <label >Only show products in stock</label>
                </div>
                
            </div>
        );
    }
}

export default SearchBar;