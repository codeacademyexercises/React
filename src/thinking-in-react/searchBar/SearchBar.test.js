import React from 'react';
import SearchBar from './SearchBar.js';
import Renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('Checking ',()=>{
    it('Snapshot test will work',()=>{
        const tree = Renderer.create(<SearchBar/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});