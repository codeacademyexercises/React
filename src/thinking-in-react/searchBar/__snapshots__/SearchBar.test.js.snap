// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking  Snapshot test will work 1`] = `
<div
  className="SearchBarContainer"
>
  <div
    className="SearchBar"
  >
    <input
      id="searchBar"
      placeholder="Search..."
      type="search"
    />
  </div>
  <div
    className="checkbox"
  >
    <input
      id="inStock"
      type="checkbox"
    />
    <label>
      Only show products in stock
    </label>
  </div>
</div>
`;
